.ci/Jenkinsfile

```
@@ -304,6 +301,26 @@ def isPrAffected(integrationName) {
   return false
 }
 
+def flattenMap(Map manifest, String separator) {
+  return manifest.collectEntries { k, v ->
+    v instanceof Map ?
+      flattenMap(v, separator).collectEntries { q, r ->
+        [(k + separator + q): r] } : [(k):v]
+  }
+}
+
+def kibanaVersionConditionFromManifest(manifest) {
+  def flattened = flattenMap(manifest, ".")
+  // echo "Manifest flattened all: ${flattened}"
+  echo "Manifest flattened version: ${flattened["conditions.kibana.version"]} - ${flattened[nonexist]}"
+
+  def version = manifest["conditions"]["kibana.version"]
+  if (version == null) {
+    version = manifest['conditions']['kibana']['version']
+  }
+  return version
```

If the map is flattened, we should be able to directly return it:
```suggestion
  return manifest['conditions.kibana.version']
```

---

Sure! I wanted to check about the value returned by `flattenMap` to perform some tests. I've already updated the function to return that value instead.